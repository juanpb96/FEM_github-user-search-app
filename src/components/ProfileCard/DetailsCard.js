export const DetailsCard = ({ repos, followers, following }) => {
    return (
        <div className='[ details ] [ bg-primary br-10 mb-24 ]'>
            <div>
                <p className='[ mb-8 fs-xsmall ]'>Repos</p>
                <p className='[ title-color fs-medium fw-700 ]'>{ repos }</p>
            </div>
            <div>
                <p className='[ mb-8 fs-xsmall ]'>Followers</p>
                <p className='[ title-color fs-medium fw-700 ]'>{ followers }</p>
            </div>
            <div>
                <p className='[ mb-8 fs-xsmall ]'>Following</p>
                <p className='[ title-color fs-medium fw-700 ]'>{ following }</p>
            </div>
        </div>
    );
};
