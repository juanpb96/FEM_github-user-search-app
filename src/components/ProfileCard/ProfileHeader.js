export const ProfileHeader = ({ atName, created_at, name }) => {

    const joinDate = new Date(created_at);
    const [ day, month, year ] = [
        joinDate.getDate(),
        joinDate.toLocaleString('en-US', { month: 'short' }),
        joinDate.getFullYear()
    ];

    return (
        <header className='[ profile__header ]'>
            <h2 
                className='[ title-color fs-medium fw-700 ]'
                datatype='username'
            >{ name ? name : atName }</h2>
            <p 
                className='[ text-color-primary fw-400 fs-small ]'
                datatype='join-date'
            >{ `Joined at ${ day } ${ month } ${ year }` }</p>
            <p 
                className='[ text-color-blue fw-400 ]'
                datatype='at'
            >@{ atName }</p>
        </header>
    );
}
