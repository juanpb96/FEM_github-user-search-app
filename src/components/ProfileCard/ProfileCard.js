import { useSelector } from "react-redux";
import { DetailsCard } from "./DetailsCard";
import { ListIconItem } from "./ListIconItem";
import { ProfileHeader } from "./ProfileHeader";

export const ProfileCard = () => {
    const { 
        login: atName,
        avatar_url: avatar,
        bio,
        blog,
        company,
        followers,
        following,
        created_at,
        location,
        name,
        public_repos: repos,
        twitter_username: twitter
    } = useSelector( state => state.user );

    const icons = [
        { name: 'location', value: location },
        { name: 'webpage', value: blog },
        { name: 'twitter', value: twitter },
        { name: 'company', value: company }
    ];

    return (
        <article className='[ profile ] [ bg-secondary card-border ]'>
            
            <ProfileHeader 
                atName={ atName }
                avatar={ avatar }
                created_at={ created_at }
                name= { name }
            />

            <p 
                className='[ profile__description ][ mb-24 text-color-secondary fs-small fw-400 lh-large ]'
            >{ bio ? bio : 'This profile has no bio' }</p>
            
            <DetailsCard 
                repos={ repos }
                followers={ followers }
                following={ following }
            />

            <ul className='[ list ] [ grid-flow ] [ no-bullets ]'>
                {
                    icons.map( item => (
                        <ListIconItem
                            key={ item.name } 
                            name={ item.name }
                            value={ item.value }
                        />
                    ))
                }
            </ul>
        </article>
    );
};
