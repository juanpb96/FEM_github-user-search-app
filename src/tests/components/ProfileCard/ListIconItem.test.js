import { shallow } from "enzyme";
import { ListIconItem } from "../../../components/ProfileCard/ListIconItem";

describe('Test <ListIconItem />', () => { 
    test('should render correctly', () => { 
        const wrapper = shallow(
            <ListIconItem 
                name={'location'}
                value={'Chicago'}
            />
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('title').text()).toBe('location');
        expect(wrapper.find('span').text()).toBe('Chicago');
    });

    test('should add an anchor tag when it is a website', () => { 
        const wrapper = shallow(
            <ListIconItem 
                name={'website'}
                value={'https://juanbonilla.me'}
            />
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('title').text()).toBe('website');
        expect(wrapper.find('a').text()).toBe('https://juanbonilla.me');
    });

    test('should add "disable" datatype and "Not Available" when there is no value', () => { 
        const wrapper = shallow(
            <ListIconItem 
                name={'company'}
                value={''}
            />
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.list__item').prop('datatype')).toBe('disabled');
        expect(wrapper.find('title').text()).toBe('company');
        expect(wrapper.find('span').text()).toBe('Not Available');
    });
});