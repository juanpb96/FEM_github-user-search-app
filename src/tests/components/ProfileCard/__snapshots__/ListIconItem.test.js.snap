// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <ListIconItem /> should add "disable" datatype and "Not Available" when there is no value 1`] = `
<li
  className="[ list__item ] [ fs-small ]"
  datatype="disabled"
>
  <div
    className="icon"
  >
    <svg
      aria-labelledby="companyIcon"
      role="img"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title
        id="companyIcon"
      >
        company
      </title>
      <g>
        <path
          d="M10.858 1.558L1.7.167A1.477 1.477 0 00.517.492 1.49 1.49 0 000 1.608v17.559c0 .458.375.833.833.833h2.709v-4.375c0-.808.65-1.458 1.458-1.458h2.083c.809 0 1.459.65 1.459 1.458V20h3.541V3a1.46 1.46 0 00-1.225-1.442zM4.583 12.292h-1.25a.625.625 0 010-1.25h1.25a.625.625 0 010 1.25zm0-2.5h-1.25a.625.625 0 010-1.25h1.25a.625.625 0 010 1.25zm0-2.5h-1.25a.625.625 0 010-1.25h1.25a.625.625 0 010 1.25zm0-2.5h-1.25a.625.625 0 010-1.25h1.25a.625.625 0 010 1.25zm4.167 7.5H7.5a.625.625 0 010-1.25h1.25a.625.625 0 010 1.25zm0-2.5H7.5a.625.625 0 010-1.25h1.25a.625.625 0 010 1.25zm0-2.5H7.5a.625.625 0 010-1.25h1.25a.625.625 0 010 1.25zm0-2.5H7.5a.625.625 0 010-1.25h1.25a.625.625 0 010 1.25zM18.85 9.035l-5.933-1.242V20h5.625A1.46 1.46 0 0020 18.542V10.46c0-.688-.47-1.274-1.15-1.425zM16.875 17.5h-1.25a.625.625 0 010-1.25h1.25a.625.625 0 010 1.25zm0-2.5h-1.25a.625.625 0 010-1.25h1.25a.625.625 0 010 1.25zm0-2.5h-1.25a.625.625 0 010-1.25h1.25a.625.625 0 010 1.25z"
        />
      </g>
    </svg>
  </div>
  <span>
    Not Available
  </span>
</li>
`;

exports[`Test <ListIconItem /> should add an anchor tag when it is a website 1`] = `
<li
  className="[ list__item ] [ fs-small ]"
>
  <div
    className="icon"
  >
    <svg
      aria-labelledby="websiteIcon"
      role="img"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title
        id="websiteIcon"
      >
        website
      </title>
    </svg>
  </div>
  <a
    href="https://juanbonilla.me"
  >
    https://juanbonilla.me
  </a>
</li>
`;

exports[`Test <ListIconItem /> should render correctly 1`] = `
<li
  className="[ list__item ] [ fs-small ]"
>
  <div
    className="icon"
  >
    <svg
      aria-labelledby="locationIcon"
      role="img"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title
        id="locationIcon"
      >
        location
      </title>
      <path
        d="M12.797 3.425C11.584 1.33 9.427.05 7.03.002a7.483 7.483 0 00-.308 0C4.325.05 2.17 1.33.955 3.425a6.963 6.963 0 00-.09 6.88l4.959 9.077.007.012c.218.38.609.606 1.045.606.437 0 .828-.226 1.046-.606l.007-.012 4.96-9.077a6.963 6.963 0 00-.092-6.88zm-5.92 5.638c-1.552 0-2.813-1.262-2.813-2.813s1.261-2.812 2.812-2.812S9.69 4.699 9.69 6.25 8.427 9.063 6.876 9.063z"
      />
    </svg>
  </div>
  <span>
    Chicago
  </span>
</li>
`;
